{{cookiecutter.project_name}}/noxfile.py

```diff
@@ -1,20 +1,66 @@
 import nox
+from pathlib import Path
+import os
 
 # Options to modify nox behaviour
+nox.options.default_venv_backend = "uv|virtualenv"
 nox.options.reuse_existing_virtualenvs = True
 
+VENV_DIR = Path("./venv").resolve()
+
 @nox.session(name="docs")
-def build_docs(session):
+def build_docs(session: nox.Session) -> None:
     """Build the documentation and load it in a browser tab, rebuilding on changes."""
-    envbindir = session.bin
     session.install("-e", ".[docs]")
     with session.chdir("docs/"):
-        session.run(
-            "sphinx-autobuild",
-            "-j",
-            "auto",
-            "--open-browser",
-            "-qT",
-            ".",
-            f"{envbindir}/../tmp/html",
-        )
+        # For local development
+        if session.interactive:
+            session.run(
+                "sphinx-autobuild",
+                "-j",
+                "auto",
+                "--open-browser",
+                "-qT",
+                ".",
+                "build/html/",
+            )
+        # For CI testing if documentation builds
+        else:
+            session.run(
+                "sphinx-build",
+                "-b",
+                "html",
+                "-W",
+                "--keep-going",
+                ".",
+                "build/html/",
+            )
+
+@nox.session(name="generated-project-tests")
```

Comment:
```suggestion
                "build/html/",
            )


@nox.session(name="generated-project-tests")
```

I have a few similar suggestions – I think we should configure Ruff to lint and ask for two lines between methods.